addons/project_timesheet_holidays/models/hr_employee.py

```diff
@@ -19,15 +19,24 @@ def create(self, vals):
 
     def write(self, vals):
         result = super(Employee, self).write(vals)
-        if 'active' in vals:
-            if vals.get('active'):
-                # Create future holiday timesheets
-                self._create_future_public_holidays_timesheets(self)
-            else:
+        if 'active' in vals or 'resource_calendar_id' in vals:
+            has_calendar_future_holidays = 'resource_calendar_id' in vals and \
+                self.env['resource.calendar.leaves'].search([
```

Comment:
```suggestion
                self.env['resource.calendar.leaves'].search_count([
```

Reply:
I think In 15.0 we don't have anything like `limit=1` for `search_count`, that's why I've used `search` Instead of `search_count` and we only care about the existence of future holidays thus we can do a regular search and limit ourselves to 1 element. :slightly_smiling_face:

Reply:
Hm, you're right: `serach_count` doesn't take a limit arg. However, I still think it is faster to count all records rather than reading all data of a record using limit 1. @xavierbol what do you think?

Reply:
@auon-odoo @arpr-odoo Good question, TBH. 🤔 
By doing the search method, we did not really read all data, we will just return a id found and that it.
I think the both solution are good in that use case. I tend to say, the search method with `limit=1` will be faster but it is a guessing. 🙂